.circleci/config.yml

```
@@ -803,6 +803,44 @@ jobs:
           command: |
             ./gradlew ui:revenuecatui:connectedDebugAndroidTest
 
+  run-maestro-paywall-tester-tests:
+    description: "Run Maestro tests for Paywall Tester app"
+    <<: *android-machine-emulator
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - install-sdkman
+      - run:
+          name: Replace API_KEY (Using Test store API key from Production Integration Tests Android project)
+          working_directory: examples/paywall-tester/src/main/java/com/revenuecat/paywallstester/
+          command: |
+            sed -i s/\"API_KEY_A\"/\"$TEST_STORE_API_KEY\"/ Constants.kt
+      - android/create_avd:
+          avd_name: test-paywall-tester
+          system_image: system-images;android-32;google_apis;x86_64
+          install: true
+      - android/start_emulator:
+          avd_name: test-paywall-tester
+          post_emulator_launch_assemble_command: ./gradlew examples:paywall-tester:assembleDebug
+      - android/restore_gradle_cache
+      - run:
+          name: Install Paywall Tester on emulator
+          command: |
+            adb install examples/paywall-tester/build/outputs/apk/debug/paywall-tester-debug.apk
+      - run:
+          name: Install Maestro
+          command: |
+            curl -Ls "https://get.maestro.mobile.dev" | bash
+            echo 'export PATH="$HOME/.maestro/bin:$PATH"' >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: Run Maestro tests
+          command: |
+            maestro test maestro_tests/paywall_tester/
+      - store_artifacts:
+          path: ~/.maestro/tests/
```

Btw this crossed my mind! Could we: 

```suggestion
            maestro test --format junit maestro_tests/paywall_tester/
      - store_artifacts:
          path: ~/.maestro/tests/
      - store_test_results:
          path: ~/.maestro/tests/ # No idea if this folder is correct, but we can use --output if needed.
```

([Docs](https://docs.maestro.dev/cli/test-suites-and-reports#generating-reports))

---

Hmm indeed, this would be a nice improvement. Great idea!